### PR TITLE
feat (ldap): add support for search paging and dereferencing aliases

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

**Review by austingebauer — APPROVED** [review]
This looks good to me! Would be good to give it a quick test in @raymonstah branch on the Vault side.

File: ldap/client.go [hunk 1]
```diff
@@ -525,15 +528,32 @@ func (c *Client) filterGroupsSearch(userDN string, username string) ([]*ldap.Ent
...
-		Attributes: []string{
-			c.conf.GroupAttr,
-		},
-		SizeLimit: math.MaxInt32,
-	})
+	var result *ldap.SearchResult
+	switch {
+	case c.conf.MaximumPageSize > 0:
+
+		result, err = c.conn.SearchWithPaging(&ldap.SearchRequest{
```
Comment (raymonstah) [inline ldap/client.go:535]: Can we extract the `ldap.SearchRequest` parameter out and reuse it across both `SearchWithPaging` and `Search`? I believe they are the same.

**Review by raymonstah — APPROVED** [review]
Looks good! Thanks for adding this 🎉

File: ldap/client.go (hunk 1 above)
Comment (jimlambrt) [reply ldap/client.go]: ty!

**Pull request merged**